V array<T>.delete not generated (no function to generate it?)

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, "Fedora release 34 (Thirty Four)"
Processor: 8 cpus, 64bit, little endian, AMD Ryzen 7 3700U with Radeon Vega Mobile Gfx
CC version: cc (GCC) 11.2.1 20210728 (Red Hat 11.2.1-1)

getwd: /home/litenite/Documents/vwork/tests
vmodules: /home/litenite/.vmodules
vroot: /home/litenite/Documents/v
vexe: /home/litenite/Documents/v/v
vexe mtime: 2021-09-06 18:52:01
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 da53f81.83d492b

Git version: git version 2.31.1
Git vroot status: 0.2.4-47-g83d492bc
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 333c14de

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg main.v`
*util/set.v*
```v 
module util

pub struct Set<T> {
	mut:
	field []T
}

pub fn (mut s Set<T>) add<T>(value T) bool {
	mut result := false
	
	if value !in s.field {
		s.field << value
		result = true
	}

	return result
}

pub fn (mut s Set<T>) remove<T>(value T) bool {
	mut result := false

	if value in s.field {
		ndx := s.field.index(value)
		println(ndx)
		// s.field = s.field.filter(it != value)// delete(ndx)
		s.field.delete(ndx)
		result = true
	}

	return result
}
```

*main.v*
```v
module main

import util as u

fn main() {
	mut set := u.Set<int>{}

	mut added := set.add(4)
	println(added)
	added = set.add(3)
	println(added)
	added = set.add(3)
	println(added)
	println(set)
	mut removed := set.remove(4)
	println(set)
}

```

**What did you expect to see?**

A mock generic set that allowed adding and removing.

true
true
false
Set[4,3]
Set[3]

**What did you see instead?**
```
==================
/tmp/v/gen_type_alias.14247993556310016351.tmp.c:10402: warning: implicit declaration of function 'Array_int_delete'
tcc: error: undefined symbol 'Array_int_delete'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

P.S. The function `Array_int_delete` is not listed anywhere else in the c file. Looking deeper, I found that `vlib/v/gen/c/array.v` has a gen_array_[fn-name] function for every operation, but does not have a gen_array_delete fn.

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -749,6 +749,18 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 		g.expr(node.args[0].expr)
 		g.write('})')
 		return
+	} else if left_sym.kind == .array && node.name == 'delete' {
+		g.write('array_delete(')
+		if node.left_type.is_ptr() {
+			g.expr(node.left)
+		} else {
+			g.write('&')
+			g.expr(node.left)
+		}
+		g.write(', ')
+		g.expr(node.args[0].expr)
+		g.write(')')
+		return
 	}
 
 	if left_sym.kind in [.sum_type, .interface_] {

--- a/vlib/v/tests/generics_array_delete_test.v
+++ b/vlib/v/tests/generics_array_delete_test.v
@@ -1,0 +1,48 @@
+struct Set<T> {
+mut:
+	field []T
+}
+
+fn (mut s Set<T>) add<T>(value T) bool {
+	mut result := false
+
+	if value !in s.field {
+		s.field << value
+		result = true
+	}
+
+	return result
+}
+
+fn (mut s Set<T>) remove<T>(value T) bool {
+	mut result := false
+
+	if value in s.field {
+		ndx := s.field.index(value)
+		s.field.delete(ndx)
+		result = true
+	}
+
+	return result
+}
+
+fn test_generics_array_delete() {
+	mut set := Set<int>{}
+
+	mut added := set.add(4)
+	println(added)
+	assert added
+
+	added = set.add(3)
+	println(added)
+	assert added
+
+	added = set.add(3)
+	println(added)
+	assert !added
+
+	println(set)
+	mut removed := set.remove(4)
+	println(removed)
+	assert removed
+}

